docs/src/catalyst_functionality/compositional_modeling.md

```diff
@@ -5,16 +5,44 @@ can construct the earlier repressilator model by composing together three
 identically repressed genes, and how to use compositional modeling to create
 compartments.
 
+## A note on *completeness*
+Catalyst `ReactionSystem` can either be *complete* or *incomplete*. When created using the `@reaction_network` DSL they are *created as complete*. Here, only complete `ReactionSystem`s can be used to create the various problem types (e.g. `ODEProblem`). However, only incomplete `ReactionSystem`s can be composed using the features described below. Hence, for compositional modeling, `ReactionSystem` must be created as incomplete, and later set to complete before simulation.
+
+To create incomplete `ReactionSystem`s using the DSL as complete, use the `@network_component` instead of `@reaction_network`:
+```@example ex0
+using Catalyst
+degradation_component = @network_component begin
+  d, X --> 0
+end
+```
+Or when created directly, use the `complete = false` argument:
+```@example ex0
+@parameters d
+@variable t
+@species X(t)
+rx = Reaction(d, [X], nothing)
+@named degradation_component = ReactionSystem([rs], t; complete = false)
+```
+We can test whether a system is complete using the `ModelingToolkit.iscomplete` function:
+```@example ex0
+ModelingToolkit.iscomplete(degradation_component)
+```
+To make a incomplete system complete, we can use the `complete` function:
+```@example ex0
+degradation_component_complete = complete(degradation_component)
+ModelingToolkit.iscomplete(degradation_component_complete)
+```
+
```

Comment:
Post v14 I would like to revisit this an write an improved description. I already have some write-up in local branches. For now, I figured it would be best to simple describe what is required. Next, we can look at the best way to actually document everything.